### DISCUSSION TITLE
Configuring texlab for neovim and evince

### DISCUSSION THREAD

**Discussion by NiklasKappel** [latex-lsp/texlab#925] [Q&A] [answered]
I just installed texlab and evince-synctex and set up the following configuration according to the wiki:

```
lspconfig.texlab.setup({
	settings = {
		texlab = {
			auxDirectory = "build",
			build = {
				args = { "-pdf", "-interaction=nonstopmode", "-synctex=1", "%f", "-auxdir=build" },
				onSave = true,
				forwardSearchAfter = true,
			},
			chktex = {
				onOpenAndSave = true,
			},
			forwardSearch = {
				executable = "evince-synctex",
				args = { "-f", "%l", "%p", '"code -g %f:%l"' },
			},
		},
	},
})
```

For me, autocompletion, diagnostics and build on save work. However, forward search does not seem to work.

After the pdf is built, evince does not open (should it?). If I open evince manually, the viewport does not move after the pdf is built.

If I call `:TexlabBuild`, neovim reports `Build Error`. If I call `:TexlabForward`, neovim reports `Search Error`.

Is there a problem with my config? How can I debug or view more information about the command errors?

**Comment by NiklasKappel** [comment, marked as answer]
I found out that my Fedora toolbox was missing the package `python3-gobject`. With this, forward search works now.

`:TexlabBuild` does not work, but I don't need it with build on save.

Reverse search does not work, but I guess this a problem with Neovim?

**Comment by pfoerster** [comment]
> :TexlabBuild does not work, but I don't need it with build on save.

Does your build finish without errors? Maybe, the error message reported by Neovim is a bit misleading here (`texlab` returns a different exit code when the build completes with errors)

> Reverse search does not work, but I guess this a problem with Neovim?

Yes, reverse search needs to be implemented by the editor and the viewer. The LSP server does not have an effect here.

**Comment by NiklasKappel** [comment]
> Does your build finish without errors? Maybe, the error message reported by Neovim is a bit misleading here (texlab returns a different exit code when the build completes with errors)

I guess this is it, thanks. Trying again, the build finishes, the document is fine but neovim reports "Build Error". I can not find any errors in my [logfiles](https://pastebin.com/3xWCDHCA) and there are no diagnostics for my complex template, but if I try a minimal example

```
\documentclass{article}
\begin{document}
text
\end{document}
```

The message is "Build Success".

The only other thing I'm missing is support for magic comments to tell texlab to use xelatex for the one project I have that requires it.

Otherwise texlab works great for me and I'm very happy with it. :)

**Comment by winstxnhdw** [comment]
Hey, I am getting the same issue here. Did you ever manage to fix this?

**Reply by NiklasKappel** [reply]
I decided not to use reverse search and this was my final texlab config
```lua
lspconfig.texlab.setup({
	settings = {
		texlab = {
			build = {
				args = { "-pdf", "-interaction=nonstopmode", "-synctex=1", "%f", "-auxdir=build" },
				auxDirectory = "build",
				forwardSearchAfter = true,
				onSave = true,
			},
			chktex = {
				onOpenAndSave = true,
			},
			forwardSearch = {
				args = { "-f", "%l", "%p", '"code -g %f:%l"' },
				executable = "evince-synctex",
			},
		},
	},
})
```

**Reply by winstxnhdw** [reply]
Sure but doesn’t nvim still tell you Build Error?

**Reply by winstxnhdw** [reply]
Also, you can use nvim-texlabconfig to help you implement reverse search easier.

**Reply by NiklasKappel** [reply]
I don't have it set up right now, sorry. IIRC build on save worked perfectly fine so I never had to use `:TexlabBuild`.

**Reply by winstxnhdw** [reply]
Oh, no. All’s good. I was just letting you know that reverse search can be done.
